test/test_client.rb

```diff
@@ -176,6 +176,13 @@ def test_find_retries_on_errors
     @client.find(:all, :foo => :bar)
   end
 
+  def test_find_waits_configured_time_before_next_request
+    @client.options[:recoverable_error_wait_secs] = 3.14
+    @client.expects(:sleep).with(3.14).times(3)
+    @client.stubs(:find_every).raises(Rets::MiscellaneousSearchError.new(0, 'Foo'))
+    @client.find(:all, :foo => :bar) rescue nil
```

Comment:
My thoughts here were that asserting on the raised error might make it less apparent that the test was about the sleep rather than the exception being raised.